cmd/kubectl/kubectlcobra/prune.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// package kubectlcobra contains cobra commands from kubectl
+package kubectlcobra
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/kubectl/pkg/cmd/apply"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/validation"
+)
+
+// PruneOptions encapsulates the necessary information to
+// implement the prune functionality.
+type PruneOptions struct {
```

Comment:
Can all these fields be made private instead of public?

Reply:
I'll make them private to start. If we need change any of them in the future--we'll do it then.